src/Mod/Arch/ArchCurtainWall.py

```diff
@@ -300,15 +303,41 @@ def execute(self,obj):
                 return
 
         facets = []
-
         faces = []
+        curtainWallBaseShapeEdges = None
+        curtainWallEdges = None
+
         if obj.Base.Shape.Faces:
             faces = obj.Base.Shape.Faces
         elif obj.Height.Value and obj.VerticalDirection.Length:
             ext = FreeCAD.Vector(obj.VerticalDirection)
             ext.normalize()
             ext = ext.multiply(obj.Height.Value)
-            faces = [edge.extrude(ext) for edge in obj.Base.Shape.Edges]
+            # ArchSketch feature :
+            if hasattr(obj.Base, 'Proxy'):
```

Comment:
This will leave out all curtain walls whose Base object has a Proxy, but no getCurtainWallBaseShapeEdgesInfo

Reply:
Thanks for checking :)  See if I understand you and below clarify the algorithm :-

- Line 307 :  set curtainWallBaseShapeEdges = None
- Line 317 :  though there is no elif/else after this, the test is in Line 320, 322, 336

- Line 320 :  it test if there is information in 'curtainWallBaseShapeEdges'
- Line 322 :  (elif) if above is not, default would following CurtainWall.OverrideEdges, if the Base is a Sketch
 (partially work, i.e. without 'Toponaming-Tolerant' support)
- Line 336 :  (else) this catch all, and it revert back to Base's edge - i.e. obj.Base.Shape.Edges

Hope this works, would review if otherwise.

Reply:
Oh you're right. I didn't look well enough :) Sorry about the noise.

I'll leave this to merge after #13783 okay?

Reply:
Thanks, feel free to proceed as you plan.

Incidentally, I like the name Arch more than BIM :)  And an add-on allow it be updated in a more free timetable not tied to the release cycle of FC itself.

Anyway, will have a look at your big merge to understand your big plan :D

Reply:
I know, I was myself not sure about which path to take for a long time... But at the end, there is duplication work happening (same things in both Arch and BIM), and also it's time the BIM stuff gets the eyes of everybody (too few contributors when it's an addon on my own github), and also everybody thinks it's a good idea to have a built-in BIM workbench, same as we just did with CAM...